packages/vite/src/node/utils.ts

```diff
@@ -1069,6 +1070,82 @@ function backwardCompatibleWorkerPlugins(plugins: any) {
   return []
 }
 
+function deepClone<T>(value: T): T {
+  if (Array.isArray(value)) {
+    return value.map((v) => deepClone(v)) as T
+  }
+  if (isObject(value)) {
+    const cloned: Record<string, any> = {}
+    for (const key in value) {
+      cloned[key] = deepClone(value[key])
+    }
+    return cloned as T
+  }
+  if (typeof value === 'function') {
+    return value as T
+  }
+  return structuredClone(value)
```

Comment:
Since arrays, objects and functions are already handled above, what cases are covered here with the `structuredClone`? From a [quick test](https://perf.link/#eyJpZCI6InhtNWliaWk2ajN6IiwidGl0bGUiOiJzdHJ1Y3R1cmVkQ2xvbmUgaW1wYWN0IiwiYmVmb3JlIjoiY29uc3QgZGF0YSA9IFsuLi5BcnJheSgxMDAwKS5rZXlzKCldIiwidGVzdHMiOlt7Im5hbWUiOiJTaW1wbGUiLCJjb2RlIjoiZGF0YS5maW5kKHggPT4geCA9PT0gMTAwKSIsInJ1bnMiOlsxMDAwLDg3NDAwMCw2MTAwMCwxNTQwMDAsMTQ4NTAwMCw2MTMwMDAsMTA3NTAwMCwxMDAwLDE0MzAwMDAsMjIyMDAwMCwxMzQwMDAsMTM3MDAwLDEwMDAwLDUwMDAsNjkwMDAsMjE5MDAwLDMwMzAwMCwzMzIwMDAsMjQxMDAwLDIwMDAsMjU4MDAwLDU5ODAwMCwzNzAwMDAsNTAwMCw1NDAwMCwxMzgwMDAsNDgwMDAsNDAwMDAsMTQ5MDAwLDY5MzAwMCwzMTAwMCw2NDEwMDAsMTM1MDAwLDM0NzAwMCwxNDEwMDAsMTcwMDAsMTEwMDAwLDE3MzAwMCw1MzcwMDAsNzAwMCwxMzAwMCwyMjcwMDAsMTQxMDAwLDEzNDAwMCwxNTgwMDAsMTM1MDAwLDEzMjAwMCwxNTIwMDAsMjUwMDAsODAwMDAsNjcwMDAsMTIwNDAwMCwyMDM5MDAwLDIzMjAwMCwxMTAwMDAsNTQwMDAsNjcwMDAsMTg5MDAwLDkwMjAwMCwxNjYwMDAsNTgwMDAsNDYwMDAsMTAyMDAwLDQyNjAwMCwyNzAwMCwxOTgwMDAsMTYyMDAwLDIxMjkwMDAsMTQwMDAsMjMzMDAwLDExNzAwMCwyNDcwMDAsNTEwMDAsMTEyMDAwLDIwMzAwMCwxNTEwMDAsNTE0MDAwLDEwODAwMCw4MTAwMCw1NDUwMDAsMTUyMDAwLDQ3MDAwLDMxMDAwMCw1MDAwLDEzMjAwMCwyMzAwMDAsMTczMDAwLDUxODAwMCw4MjAwMCwyMDAwLDE5MDAwMCwzNjIwMDAsMjA0MDAwLDI1ODAwMCwyMDAwLDEyNzAwMCwxNDQwMDAsMTAzMDAwLDU2MjAwMCwzMTYwMDBdLCJvcHMiOjI5NTI4MH0seyJuYW1lIjoic3R1Y3R1cmVkQ2xvbmUiLCJjb2RlIjoiZGF0YS5maW5kKHggPT4gc3RydWN0dXJlZENsb25lKHgpID09PSAxMDApIiwicnVucyI6WzIwMDAsMzAwMCwyNTAwMCwxNjAwMCwyNDAwMCwyNjAwMCwyNDAwMCwzMTAwMCwyMTAwMCwzODAwMCw1MDAwLDQwMDAsMTAwMCwxMDAwLDMwMDAsODAwMCwxMjAwMCw2MDAwLDQwMDAsMzAwMCwxMTAwMCw1MDAwLDEyMDAwLDYwMDAsNjAwMCw1MDAwLDcwMDAsNjAwMCw5MDAwLDE4MDAwLDMwMDAsNTAwMCwxMjAwMCwxMDAwMCwxMjAwMCwyMDAwLDYwMDAsNzAwMCwxNjAwMCwyMDAwLDUwMDAsMTAwMDAsNDAwMCw1MDAwLDEyMDAwLDE5MDAwLDcwMDAsNzAwMCw4MDAwLDExMDAwLDUwMDAsMjQwMDAsMzYwMDAsMTAwMDAsMTMwMDAsNDAwMCw0MDAwLDgwMDAsNDAwMCw4MDAwLDcwMDAsMTAwMDAsNDAwMCwxNTAwMCw4MDAwLDEwMDAwLDUwMDAsMjcwMDAsMTQwMDAsNjAwMCw2MDAwLDYwMDAsMjAwMCw1MDAwLDgwMDAsNjAwMCwxNDAwMCwxMDAwLDcwMDAsOTAwMCw3MDAwLDgwMDAsODAwMCwzMDAwLDYwMDAsMTEwMDAsMTIwMDAsOTAwMCwxMDAwMCwxMDAwLDcwMDAsNzAwMCw5MDAwLDkwMDAsMTQwMDAsMzAwMCw3MDAwLDgwMDAsOTAwMCw4MDAwXSwib3BzIjo5NDcwfV0sInVwZGF0ZWQiOiIyMDI0LTExLTEyVDA2OjQzOjA1LjAyOFoifQ%3D%3D), `structuredClone` seems to have a big impact when used on basic primitives.

Reply:
It was mainly for `RegExp`. I put the `structuredClone` in a condition 👍 (38162a0a1603da5de06d683d97a1b175289caeeb, e4364e59fbd760c2f7f6e63a2cc137720cd2690d).

Reply:
I don't think it's likely for someone to mutate a `RegExp`, but I don't mind having the code for it either ways.